recipe/meta.yaml

```diff
@@ -2,37 +2,47 @@
 {% set version = "0.55.0" %}
 
 {% set name = goname.split('/')[-1] %}
-{% set pkg_src = ('src/'+goname).replace("/",os.sep) %}
+{% set pkg_src = ('src/'+goname).replace("/",os.sep) %}  # src/github.com/junegunn/fzf
 
 package:
   name: {{ name|lower }}
   version: {{ version|replace("-", "_") }}
 
 source:
-  - folder: {{ pkg_src }}
-    url: https://{{ goname }}/archive/v{{ version }}.tar.gz
-    sha256: 805383f71bca7f8fb271ecd716852aea88fd898d5027d58add9e43df6ea766da
+  folder: {{ pkg_src }}
+  url: https://{{ goname }}/archive/v{{ version }}.tar.gz
+  sha256: 805383f71bca7f8fb271ecd716852aea88fd898d5027d58add9e43df6ea766da
 
 build:
   number: 0
+  skip: true  # [s390x]
   script:
     - pushd {{ pkg_src }}
-    - go build -ldflags "-X main.revision=conda-forge" -v -o $PREFIX/bin/fzf  # [not win]
-    - go build -ldflags "-X main.revision=conda-forge" -v -o %LIBRARY_BIN%\fzf.exe  # [win]
+    - go build -ldflags "-X main.revision=anaconda" -v -o $PREFIX/bin/fzf  # [not win]
+    - go build -ldflags "-X main.revision=anaconda" -v -o %LIBRARY_BIN%\fzf.exe  # [win]
     - bash -c "mkdir -p ${PREFIX}/share/fzf/shell; cp -v shell/* ${PREFIX}/share/fzf/shell/"
     - bash -c "mkdir -p ${PREFIX}/share/fzf/plugin; cp -v plugin/* ${PREFIX}/share/fzf/plugin/"
-    - go-licenses save . --save_path ../../../../library_licenses
+    - bash -c "go-licenses save . --save_path ${SRC_DIR}/library_licenses"
     # Clear out cache to avoid file not removable warnings
     - chmod -R u+w $(go env GOPATH) && rm -r $(go env GOPATH)  # [unix]
 
 requirements:
   build:
-    - {{ compiler('go') }}
+    - go-nocgo >=1.20
     - go-licenses
 
 test:
+  source_files:
+    - library_licenses
```

Comment:
this copies files from the work directory to the test directory - so if you do this, you're not testing whether the files are in the package

Reply:
Wouldn't that technically be true for all of the tests for this package since the files from `${PREFIX}` are no longer the same file handle as the ones inside the .bz2 file?

I did attempt to test the `library_licenses` within the `${PREFIX}`; however, it seems they never get copied there. I'm not 100% certain but it would seem conda-build includes them directly from their given paths in the `about` section into the various packages. After discovering this it seemed my options were to either attempt to install them into the `${PREFIX}` path initially, so that the tests would all take place from the same parent directory, or simply test the files that contributed to the package directly. Since that's essentially what's being done with the tests taking place on the files in the `${PREFIX}` anyway it made sense to me to verify their existence in this way.

Reply:
> Wouldn't that technically be true for all of the tests for this package since the files from ${PREFIX} are no longer the same file handle as the ones inside the .bz2 file?

Er I'm not quite following, could you clarify? [source_files](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-files) allows you to use files that are in the sdist or github repo but that haven't been included in your conda package.

Anyway, oh yeah, they're license files and the license_file entry in the about section should handle them and they won't be put in the prefix upon installation. I don't think there's a neat way to test this, at least that I can think of, but you could download and unzip a tarball and check manually. Suggest you just remove the test for them.

Reply:
also, the usual pattern is to do `if not exist` on windows rather than `bash -c` on all platforms, see [here](https://github.com/AnacondaRecipes/xz-feedstock/blob/e4727181dd6dd975f850faa66a9233fd6773eac8/recipe/meta.yaml#L41) for an example. I'm guessing that's because we want the feedstock to work for users on windows platforms who haven't got gnu tools in their toolchain somehow (e.g. via cygwin, which is the case on our CI builders)

Reply:
well, then you have to do it in the script section too, eg `mkdir %PREFIX%\share\fzf\shell && xcopy /s /y shell\* %PREFIX%\share\fzf\shell\`, maybe if it works then great but if it becomes a hassle then we could leave as-is, there are a few feedstocks that do `bash -c`

Reply:
> Er I'm not quite following, could you clarify?

Since `${PREFIX}` is just the staging area that the package gets its contents from and these licenses are pulled directly from the source their location in the source could be considered the staging area for the licenses files. In that respect, testing for non-license file existence in the staging area of `${PREFIX}` seems equivalent to testing for license file existence in the license staging area in the source. To truly test whether or not the package contains these files, the package would have to be untarred and the contents inspected. I made the assumption that verifying the contents from the package staging area was sufficient to assume these files actually made it into the resulting package archive. I'll remove the test for the licenses.

> we want the feedstock to work for users on windows platforms who haven't got gnu tools in their toolchain

> here are a few feedstocks that do bash -c

For this particular package, we should be safe to assume Windows users will have access to `bash` since this package depends upon `go-licenses` which depends upon `posix  # [win]`.

Reply:
ok, I guess in that sense they are both "staging areas", but the difference between them is that conda-build will automatically package stuff in the prefix but won't do the same with the source (work) directory. Also, when a package is being tested via instructions in the `test` section, it doesn't test stuff in the prefix that's used during build (the host environment), it installs the built package into a _further_ environment and tests there. So that it's a valid test for the actual package.

with the posix thing, if it's required in this recipe it should be stated here rather than transitively requiring it (what happens then if for some reason go-licenses drops that requirement)? My preference would be to use the more standard pattern, though, but both will obv work